core-java/src/main/java/org/spine3/server/procman/ProcessManager.java

```diff
@@ -0,0 +1,296 @@
+     */
+    private void init() {
+        if (!this.initialized) {
+            final Registry registry = Registry.getInstance();
+            final Class<? extends ProcessManager> pmClass = getClass();
+            if (!registry.contains(pmClass)) {
+                registry.register(pmClass);
+            }
+            commandHandlers = registry.getCommandHandlers(pmClass);
+            eventHandlers = registry.getEventHandlers(pmClass);
+            this.initialized = true;
+        }
+    }
+
+    /**
+     * Dispatches a command to the command handler method of the process manager.
+     *
+     * @param command the command to be executed on the process manager
+     * @param context of the command
+     * @throws InvocationTargetException if an exception occurs during command dispatching
+     */
+    protected List<EventRecord> dispatchCommand(Message command, CommandContext context) throws InvocationTargetException {
+        //noinspection DuplicateStringLiteralInspection
+        checkNotNull(command, "command");
+        checkNotNull(context, "command context");
+
+        init();
+        final Class<? extends Message> commandClass = command.getClass();
+        final Method method = commandHandlers.get(commandClass);
+        if (method == null) {
+            throw missingCommandHandler(commandClass);
+        }
+        final CommandHandlerMethod commandHandler = new ProcessManagerCommandHandler(this, method);
+        final List<? extends Message> events = commandHandler.invoke(command, context);
+        final List<EventRecord> eventRecords = toEventRecords(events, context.getCommandId());
+        return eventRecords;
+    }
+
+    private List<EventRecord> toEventRecords(final List<? extends Message> events, final CommandId commandId) {
+        return Lists.transform(events, new Function<Message, EventRecord>() {
+            @Nullable // return null because an exception won't be propagated in this case
+            @Override
+            public EventRecord apply(@Nullable Message event) {
+                if (event == null) {
+                    return EventRecord.getDefaultInstance();
+                }
+                final EventContext eventContext = createEventContext(commandId, event, getState(), whenModified(), getVersion());
+                final EventRecord result = Events.createEventRecord(event, eventContext);
+                return result;
+            }
+        });
+    }
+
+    /**
+     * Dispatches an event to the event handler method of the process manager.
+     *
+     * @param event the event to be handled by the process manager
+     * @param context of the event
+     * @throws InvocationTargetException if an exception occurs during event dispatching
+     */
+    protected void dispatchEvent(Message event, EventContext context) throws InvocationTargetException {
+        checkNotNull(event, "event");
+        checkNotNull(context, "event context");
```

Comment:
Change the string to just "context". It's shorter and removes duality in reading (now both strings start with 'event').